### DISCUSSION TITLE
Профилирование Encounter.

### DISCUSSION THREAD

**Discussion by Vrodvi** [fhir-ru/core#139] [Клиническая РГ]
По решению с совещания http://fhir-ru.zendoc.me/meetings.2022-04-26 предлагается внести следующие изменения в профилирование Encounter:

# 1)
Предлагается разделить данный профиль на два отдельных профиля:

- core-профиль в разделе [Ядро (core)](http://fhir-ru.zendoc.me/core)
  - в этом профиле, как и во всех корневых профилях, рекомендуется:
    -  обозначить только минимально необходимые ограничения, связанные с российской спецификой):
      - ссылку на пациента заменить на ссылку на КорневогоПациента
      - ссылки на организации заменить на ссылки на КорневыеОрганизации
  - перевести на русский язык все названия и определения, чтобы этот перевод наследовался далее в более специфические профили
    -  если нет ресурсов на полный перевод, то можно ограничиться переводом только обязательных элементов (с кардинальностью [1.. ] status и class), а также элементов, для которых накладываются ограничения (в данном случае - элементы со ссылками на пациента и организации, например subject, serviceProvider, hospitalization.origin\destination )

- профиль в разделе [Указание диагноза в рамках случая](http://fhir-ru.zendoc.me/diag.DsEncounter)
  - в этом профиле:
    -  указать, что он наследуется от корневого профиля
    - добавить ограничения на элемент Encounter.diagnisis в соответствии с требованиями из [гугл-таблицы](https://docs.google.com/spreadsheets/d/1WDC22c-qnWYEJZmdKcfuIqz_OvZaC_z4b5Vg5R5YSnI/edit#gid=371194862):
      - определение трёх "срезов" (основной, осложнение, сопутствующий)
        - добавление "расширений" (тип и рубрика) в эти срезы
      - перевод на русский названий и описаний для атрибутов:
        - condition (должна быть ссылка на профиль нозологической единицы)
        - use
        - rank
      - добавить правила проверки "сочетаний" рубрик (например, что если нет основного заболевания, то не должно быть сопутствующих)

# 2)
Для упрощения понимания правил проверки "сочетаний" рубрик необходимо подготовить набор примеров, демонстрирующих допустимые и недопустимые (с пояснениями причины невалидности) сочетания 

> Невалидные примеры пока что решили не делать, чтобы не путать читателей руководства

Примеры будут добавляться сюда по мере готовности: см. файл [CoreBundleEncounterWithDiagnosisCollectionInstance.txt](https://github.com/fhir-ru/core/files/8762708/CoreBundleEncounterWithDiagnosisCollectionInstance.txt)

Ранее подготовленный предварительный вариант с примерами в формате XML и Schematron: см. файл 
[старый вариант.txt](https://github.com/fhir-ru/core/files/8761484/default.txt)

# 3) 
Требуется перепроверить перечень и наименования профилей\справочников\наборов значений, определённых для данных профилей,
т.к. имеются расхождения с принятым [регламентом](http://fhir-ru.zendoc.me/conventions).
Некоторые выявленные проблемы:
- наименование расширений:
  - сейчас
    - http://fhir.ru/core/ex/core-diagnosis-type
    - http://fhir.ru/core/ex/core-diagnosis-rubric
  - должно быть 
    - http://fhir.ru/core/sd/core-ex-diagnosis-type
    - http://fhir.ru/core/sd/core-ex-diagnosis-rubric
- справочники и наборы значений:
  - рубрики (Виды нозологических единиц диагноза)
    - справочник:
      - сейчас: 
        - http://fhir.ru/core/systems/diagnosis-nosology-kinds
        - названо не по регламенту и в названии не отображен источник справочника - НСИ
        - множественное число в названии лучше не использовать (добавлено по [комменту](https://github.com/fhir-ru/core/discussions/139#discussioncomment-2653309))
      - назвать желательно:
        - http://fhir.ru/core/cs/core-cs-nsi-diagnosis-nosology-kind
    - набор значений:
      - сейчас:
        - CoreVsDiagnosisRubric и http://fhir.ru/core/vs/core-diagnosis-type
      - нужно: 
        - CoreVsDiagnosisRubric и http://fhir.ru/core/vs/core-vs-diagnosis-rubric
      - также для обозначения "срезов" (slice) необходимо определить три дополнительных набора значения:
        - CoreVsDiagnosisRubricMain и http://fhir.ru/core/vs/core-vs-diagnosis-rubric-main
        - CoreVsDiagnosisRubricCompl и http://fhir.ru/core/vs/core-vs-diagnosis-rubric-compl
        - CoreVsDiagnosisRubricSupl и http://fhir.ru/core/vs/core-vs-diagnosis-rubric-supl
  - типы (Степень обоснованности диагноза)
    - справочник:
      - сейчас: 
        - http://fhir.ru/core/systems/diagnosis-justification-degree
        - названо не по регламенту и в названии не отображен источник справочника - НСИ, 
      - назвать желательно:
        - http://fhir.ru/core/cs/core-cs-nsi-diagnosis-justification-degree
    - набор значений:
      - сейчас:
        - CoreVsDiagnosisType и http://fhir.ru/core/vs/core-diagnosis-type
      - нужно: 
        - CoreVsDiagnosisType и http://fhir.ru/core/vs/core-vs-diagnosis-type

# 4) 
Дополнительно:
- 4.1) для лучшего понимания требований этого профиля решено дополнительно оформить их в простом текстовом формате.
Описание будет добавлено тут по мере готовности:
```
<TODO>
```
- 4.2) профили на данный момент оформляются в специальном ZEN-формате, для которого отсутствует возможность конвертации в стандартный fhir-формат - StructureDefinition.
Необходимо дополнительное обсуждение на тему формирования StructureDefinition как конечного результата профилирования. Без формирования StructureDefinition мы не сможем заявлять о соответствии стандарту, а также не сможем использовать сторонние стандартные инструменты, заточенные на работу с форматом StructureDefinition.
- 4.3) в вопросе оформления "срезов" (slice) для атрибутов FHIR-ресурсов у сообщества имеются определённые сложности. Но из-за того что используется нестандартный формат описания профилей, эти вопросы, скорее всего, не будут прояснены. В связи с этим, желательно завести дополнительную дискуссию по вопросам реализации "срезов".
К примеру, для диагнозов в рамках случая требуются срезы, определяемые набором кодов, а не одиночным конкретным кодом - нужно понять каким образом такого рода срезы должны оформляться в FHIR-профилях (сейчас в ZEN-формате придуман некий альтернативный механизм и не ясно насколько он коррелирует с подходом FHIR из-за чего возникают сомнения в возможности переноса ZEN-профилей в стандартный формат StructureDefinition).

**Comment by Vrodvi** [comment]
@eugenekogan-spb @VarvaraSemenova 
Это обсуждение по поводу вот этих пунктов http://fhir-ru.zendoc.me/meetings.2022-04-26:
> 5. Предложить иерархию профилей Encounter для Core и для процесса ведения клинического диагноза Clinical IG – Родионов.
> 7. Сделать дискуссию о slicing (как вводить ограничения, как обеспечить совместимость с StructureDefnition) – Родионов.)

Пока что одна дискуссия, при необходимости выделим отдельную позже.

**Comment by JustAsWalker** [comment]
> в вопросе оформления "срезов" (slice) для атрибутов FHIR-ресурсов у сообщества имеются определённые сложности.
> К примеру, для диагнозов в рамках случая требуются срезы, определяемые набором кодов, а не одиночным конкретным кодом - нужно понять каким образом такого рода срезы должны оформляться в FHIR-профилях (сейчас в ZEN-формате придуман некий альтернативный механизм и не ясно насколько он коррелирует с подходом FHIR из-за чего возникают сомнения в возможности переноса ZEN-профилей в стандартный формат StructureDefinition).

В описании дискриминатора для слайсов - https://www.hl7.org/fhir/profiling.html#discriminator - есть такое утверждение:
> Each slice must use the [element definition](https://www.hl7.org/fhir/elementdefinition.html) for the element(s) in the discriminator(s) to ensure that the slices are clearly differentiated by assigning an appropriate value domain, depending on the discriminator type. If the type is value, or pattern, then the element definition must use either:
> 1. [ElementDefinition.fixed[x]](https://www.hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition.fixed_x_), or
> 2. [ElementDefinition.pattern[x]](https://www.hl7.org/fhir/elementdefinition-definitions.html#ElementDefinition.pattern_x_), or
> 3. **_if the element has a terminology binding, a required binding with a [Value Set](https://www.hl7.org/fhir/valueset.html#required) that enumerates the list of possible codes in the value set ("extensional definition")_**

Но так как примеры в этом описании отсутствуют, не до конца понятно, как именно это должно выглядеть и работать.

**Reply by Vrodvi** [reply]
Спасибо, похоже, что нам подходит реализация чрез тип pattern, его третий вариант - через привязку некого атрибута среза к набору значений (т.е. для каждого среза нужно определить свой набор значений).

Подготовил в формате FHS пример с использованием этого варианта:
```
Profile: CoreEncounter
Parent: Encounter
Id: core-encounter
Title: "Случай обслуживания"
Description: "Случай обслуживания."
* ^url = "http://fhir.ru/core/sd/core-encounter"

* diagnosis MS
* diagnosis.condition MS
* diagnosis.use MS
* diagnosis.rank MS

* diagnosis ^slicing.discriminator.type = #pattern
* diagnosis ^slicing.discriminator.path = "extension[diagnosisRubric].valueCodeableConcept"
* diagnosis ^slicing.description = "Срез на основе вхождения diagnosis.extension[diagnosisRubric].valueCodeableConcept в один из наборов значений."
* diagnosis ^slicing.rules = #open

* diagnosis contains
    main 1..* MS and
    compl 0..* MS and
    supl 0..*

* diagnosis[main].condition 1..1 MS
* diagnosis[main].use 0..1 MS
* diagnosis[main].rank 0..1 MS
* diagnosis[main].extension contains
    CoreExDiagnosisType named diagnosisType 0..1 MS and
    CoreExDiagnosisRubric named diagnisisRubric 1..1 MS
* diagnosis[main].extension[CoreExDiagnosisType].valueCodeableConcept 1..1 MS
* diagnosis[main].extension[CoreExDiagnosisType].valueCodeableConcept from http://fhir.ru/core/vs/core-vs-diagnosis-type (required)
* diagnosis[main].extension[CoreExDiagnosisRubric].valueCodeableConcept 1..1 MS
* diagnosis[main].extension[CoreExDiagnosisRubric].valueCodeableConcept from http://fhir.ru/core/vs/core-vs-diagnosis-rubric-main (required)

* diagnosis[compl].condition 1..1 MS
* diagnosis[compl].use 0..1 MS
* diagnosis[compl].rank 0..1 MS
* diagnosis[compl].extension contains
    CoreExDiagnosisType named diagnosisType 0..1 MS and
    CoreExDiagnosisRubric named diagnisisRubric 1..1 MS
* diagnosis[compl].extension[CoreExDiagnosisType].valueCodeableConcept 1..1 MS
* diagnosis[compl].extension[CoreExDiagnosisType].valueCodeableConcept from http://fhir.ru/core/vs/core-vs-diagnosis-type (required)
* diagnosis[compl].extension[CoreExDiagnosisRubric].valueCodeableConcept 1..1 MS
* diagnosis[compl].extension[CoreExDiagnosisRubric].valueCodeableConcept from http://fhir.ru/core/vs/core-vs-diagnosis-rubric-compl (required)

* diagnosis[supl].condition 1..1 MS
* diagnosis[supl].use 0..1 MS
* diagnosis[supl].rank 0..1 MS
* diagnosis[supl].extension contains
    CoreExDiagnosisType named diagnosisType 0..1 MS and
    CoreExDiagnosisRubric named diagnisisRubric 1..1 MS
* diagnosis[supl].extension[CoreExDiagnosisType].valueCodeableConcept 1..1 MS
* diagnosis[supl].extension[CoreExDiagnosisType].valueCodeableConcept from http://fhir.ru/core/vs/core-vs-diagnosis-type (required)
* diagnosis[supl].extension[CoreExDiagnosisRubric].valueCodeableConcept 1..1 MS
* diagnosis[supl].extension[CoreExDiagnosisRubric].valueCodeableConcept from http://fhir.ru/core/vs/core-vs-diagnosis-rubric-supl (required)


Extension: CoreExDiagnosisRubric
Id: core-ex-diagnosis-rubric
Title: "Рубрика диагноза"
Description: "Рубрика диагноза."
* ^url = "http://fhir.ru/core/sd/core-ex-diagnosis-rubric"
* value[x] 1..1 MS
* value[x] only CodeableConcept
* value[x] from http://fhir.ru/core/vs/core-vs-diagnosis-rubric (required)

Extension: CoreExDiagnosisType
Id: core-ex-diagnosis-type
Title: "Тип диагноза"
Description: "Тип диагноза."
* ^url = "http://fhir.ru/core/sd/core-ex-diagnosis-type"
* value[x] 1..1 MS
* value[x] only CodeableConcept
* value[x] from http://fhir.ru/core/vs/core-vs-diagnosis-type (required)


ValueSet: CoreVsDiagnosisType
Id:       core-vs-diagnosis-type
Title: "Типы диагноза"
Description:   "Типы диагноза."
* ^url =  "http://fhir.ru/core/vs/core-vs-diagnosis-type"


ValueSet: CoreVsDiagnosisRubricMain
Id:       core-vs-diagnosis-rubric-main
Title: "Рубрики диагноза - основные"
Description:   "Рубрики диагноза - основные."
* ^url =  "http://fhir.ru/core/vs/core-vs-diagnosis-rubric-main"


ValueSet: CoreVsDiagnosisRubricCompl
Id:       core-vs-diagnosis-rubric-compl
Title: "Рубрики диагноза - сопутствующие"
Description:   "Рубрики диагноза - сопутствующие."
* ^url =  "http://fhir.ru/core/vs/core-vs-diagnosis-rubric-compl"


ValueSet: CoreVsDiagnosisRubricSupl
Id:       core-vs-diagnosis-rubric-supl
Title: "Рубрики диагноза - осложнения"
Description:   "Рубрики диагноза - осложнения."
* ^url =  "http://fhir.ru/core/vs/core-vs-diagnosis-rubric-supl"
```

Проверено в https://fshschool.org/FSHOnline/#/ - представленный пример, по крайней мере, не даёт ошибок, так что возможно это правильный вариант реализации.
Также сформирована страница с данным профилем для демонстрации как это может выглядеть после конвертации в StructureDefinition - https://docs.miramedix.ru/fhir/IG/StructureDefinition-core-encounter.html.

**Comment by JustAsWalker** [comment]
> рубрики (Виды нозологических единиц диагноза)
> справочник:
> сейчас:
> http://fhir.ru/core/systems/diagnosis-nosology-kinds
> названо не по регламенту и в названии не отображен источник справочника - НСИ,
> назвать желательно:
> http://fhir.ru/core/cs/core-cs-nsi-diagnosis-nosology-kinds

Есть общее предложение не использовать в идентификации справочников множественное число. Все остальные перечисленные в описании объекты его не используют.

**Reply by Vrodvi** [reply]
Поддерживаю, поправил в описании.

**Comment by Vrodvi** [comment]
Дополнил в описании пункт "2)" - как временный вариант подготовлены схематичные примеры в формате XML с реализацией проверок в формате Schematron.
